core/include/tee/tee_fs.h

```diff
@@ -41,23 +41,32 @@ struct tee_fs_dirent {
 	char *d_name;
 };
 
+struct tee_fs_dir;
+struct tee_file_handle;
+
 /*
  * tee_fs implemets a POSIX like secure file system with GP extension
  */
 struct tee_file_operations {
-	int (*open)(TEE_Result *errno, const char *file, int flags, ...);
-	int (*close)(int fd);
-	int (*read)(TEE_Result *errno, int fd, void *buf, size_t len);
-	int (*write)(TEE_Result *errno, int fd, const void *buf, size_t len);
-	tee_fs_off_t (*lseek)(TEE_Result *errno,
-			      int fd, tee_fs_off_t offset, int whence);
-	int (*rename)(const char *old, const char *new);
-	int (*unlink)(const char *file);
-	int (*ftruncate)(TEE_Result *errno, int fd, tee_fs_off_t length);
+	TEE_Result (*open)(const char *name, struct tee_file_handle **fh);
+	TEE_Result (*create)(const char *name, bool overwrite,
+			     struct tee_file_handle **fh);
+	void (*close)(struct tee_file_handle **fh);
```

Comment:
I did that first, but discovered that it was more convenient to pass a reference as `close()` then can assign `NULL` to `*fh`.